pkg/controller/recoveries.go

```diff
@@ -151,9 +151,31 @@ func (c *StashController) runRecoveryInjector(key string) error {
 	if !exists {
 		// Below we will warm up our cache with a Recovery, so that we will see a delete for one d
 		fmt.Printf("Recovery %s does not exist anymore\n", key)
-	} else {
-		d := obj.(*api.Recovery)
-		fmt.Printf("Sync/Add/Update for Recovery %s\n", d.GetName())
+		return nil
 	}
+
+	d := obj.(*api.Recovery)
+	fmt.Printf("Sync/Add/Update for Recovery %s\n", d.GetName())
+	return c.runRecoveryJob(d)
+}
+
+func (c *StashController) runRecoveryJob(rec *api.Recovery) error {
```

Comment:
Where are we deleting the Job and its pods? We don't want pods to remain in success status.